### PR TITLE
Improve config file documenting comments

### PR DESCRIPTION
**Pull request by rhansen**
* Fix misleading `unprivileged-user` and `chroot-dir` documentation (leaving these values unset does not disable their respective features).
  * Document which files are read/created before chrooting and which are after.

### CODE REVIEW

File: data/postsrsd.conf.in [hunk 1]
```diff
@@ -152,22 +156,30 @@ hash-minimum = 4
...
 # Execute PostSRSd as unprivileged user
-# If you set this highly recommended option, PostSRSd will drop root
-# privileges and switch to the configured user before it enters the main loop
-# to handle untrusted input.
+# Drop root privileges and run as this user before entering the main loop and
+# handling untrusted input. To prevent PostSRSd from changing users, set this to
+# the empty string.
 #
-# Example:
+# Default if unset:
```
Comment (roehling) [inline data/postsrsd.conf.in:163]: I don't like the "Default if unset" extra paragraphs. If a default is important to document, I'd rather have it as part of the description text. Also, I have intentionally and explicitly set all values to their default values in the example config, which serves the same purpose.

File: data/postsrsd.conf.in [hunk 2]
```diff
@@ -152,22 +156,30 @@ hash-minimum = 4
...
+# layer of protection against the exploitation of security bugs in PostSRSd. To
+# prevent PostSRSd from chrooting, set this to the empty string.
 #
-# Example:
+# Default if unset:
+#     chroot-dir = "@POSTSRSD_CHROOTDIR@"
+#
+# Examples:
 #     chroot-dir = "@CMAKE_INSTALL_FULL_LOCALSTATEDIR@/lib/@PROJECT_NAME@"
+#     chroot-dir = "" # Don't chroot.
```
Comment (roehling) [inline data/postsrsd.conf.in:182]: This looks very awkward to me. Maybe we can explain at the beginning of the file how to set an empty string, i.e. `= ""`. The description already documents what an empty value means, no need to repeat it as a comment.

**Review by roehling — CHANGES_REQUESTED** [review]
I like the changes in general, but see the two comments I made.

File: data/postsrsd.conf.in (hunk 1 above)
Comment (rhansen) [reply data/postsrsd.conf.in]: Done.

File: data/postsrsd.conf.in (hunk 2 above)
Comment (rhansen) [reply data/postsrsd.conf.in]: Done.

**Pull request merged**